src/importer.js

```diff
@@ -0,0 +1,50 @@
+const fs = require('fs');
+const EventEmitter = require('events').EventEmitter;
+const csvjson = require('csvjson');
+
+const checkedFiles = [];
+
+class Importer extends EventEmitter {
+  constructor(dirWatcher, path) {
+    super();
+    // sync call:
+    // dirWatcher.on('data change', () =>
+    //   this.importSync(path)
+    // );
+    // ---------------
+    // async call:
+    dirWatcher.on('data change', () =>
+      this.import(path)
+        .then(data => console.log(data))
+        .catch(err => console.log(err)));
+  }
+
+  import(path) {
+    return new Promise((res, rej) => {
+      const files = fs.readdirSync(path);
+      const processedData = [];
+      files.forEach((file) => {
+        if (!checkedFiles.includes(file)) {
+          const data = fs.readFileSync(`./data/${file}`, { encoding: 'utf8' });
+          processedData.push(csvjson.toObject(data));
+          checkedFiles.push(file);
+          res(processedData);
```

Comment:
you resolve Promise on first retrieved item, but you should do it only after all files will be processed